datalad/distribution/tests/test_publish.py

```diff
@@ -61,7 +74,8 @@ def test_publish_simple(origin, src_path, dst_path):
 
     # don't fail when doing it again
     res = publish(dataset=source, to="target")
-    eq_(res, ([source], []))
+    # and nothing is pushed
+    eq_(res, ([], []))
```

Comment:
not that I am against such a change (return only published items) but note that this would diverge from how/what e.g. `install` and probably other API commands return -- they are trying to fulfill the promise, i.e. return requested item to be installed even if it was installed before already.  The reason for that was I believe to be able to have e.g. a Python script which you could just rerun and use those returned objects.
I wonder now if we should introduce here and elsewhere some additional option (e.g. `--affected-only`) which would switch between two behaviors. not sure though if that is feasible or not.